packages/jsii-pacmak/lib/targets/go/types/interface.ts

```diff
@@ -65,8 +65,9 @@ export class Interface extends GoType {
   public get dependencies(): Package[] {
     return [
       ...this.extendsDependencies,
-      ...getFieldDependencies(this.methods),
-      ...getFieldDependencies(this.properties),
+      ...getMemberDependencies(this.methods),
+      ...getParamDependencies(this.methods),
```

Comment:
Alternatively, you can compute this on the `InterfaceMethod` class itself. Add a `dependencies` field to the `InterfaceMethod` class, and return an array with all param dependencies and the return type dependencies for a single method. Then here you just `flatMap` the method dependencies array into the parents dependencies array.

The reason that `getFieldDependencies` is used is that a property can only have a single type reference, so adding a `dependencies` array field on a property doesn't make sense because its basically just the `reference` field.  However, an `InterfaceMethod` can have many dependencies, so collecting them one level above where the `GoTypeReference`s are is what is easiest. IE, in the `InterfaceMethod` class.

This can also be done for the `GoMethod` class for class methods. An abstract base class may be useful here.

Reply:
Is this a blocker for you?

Reply:
Not a blocker, but may be simpler/easier for you to implement this way. Either way though.

Reply:
I like this idea in general, but since that would mean having to do it both on interfaceMethod as well as ClassMethod I think it's simplest/DRYest for now to keep it as a utility function that both can use. We can always refactor later!